src/test/test-unit-name.c

```diff
@@ -199,12 +199,11 @@ static void test_unit_name_mangle(void) {
 }
 
 static int test_unit_printf(void) {
-        Manager *m = NULL;
+        _cleanup_free_ char *mid = NULL, *bid = NULL, *host = NULL, *uid = NULL, *user = NULL, *shell = NULL, *home = NULL;
+        _cleanup_free_ Manager *m = NULL;
```

Comment:
@keszybz, thank you very much for working on this!

This one should probably be `_cleanup_(manager_freep) Manager *m = NULL`. Otherwise, `asan` reports a lot of memory leaks:
```sh
=================================================================
==4855==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 200 byte(s) in 1 object(s) allocated from:
    #0 0x7fd6cd544850 in malloc (/lib64/libasan.so.4+0xde850)
    #1 0x7fd6cce5cb20 in malloc_multiply ../src/basic/alloc-util.h:74
    #2 0x7fd6cce5d17a in strv_copy ../src/basic/strv.c:106
    #3 0x564ecaaeb430 in manager_default_environment ../src/core/manager.c:571
    #4 0x564ecaaec7d5 in manager_new ../src/core/manager.c:730
    #5 0x564ecaadfdae in test_unit_printf ../src/test/test-unit-name.c:215
    #6 0x564ecaae6e7d in main ../src/test/test-unit-name.c:486
    #7 0x7fd6cb233009 in __libc_start_main (/lib64/libc.so.6+0x21009)
...
Indirect leak of 2234 byte(s) in 24 object(s) allocated from:
    #0 0x7fd6cd4dd238 in __interceptor_strdup (/lib64/libasan.so.4+0x77238)
    #1 0x7fd6cce5d1d7 in strv_copy ../src/basic/strv.c:112
    #2 0x564ecaaeb430 in manager_default_environment ../src/core/manager.c:571
    #3 0x564ecaaec7d5 in manager_new ../src/core/manager.c:730
    #4 0x564ecaadfdae in test_unit_printf ../src/test/test-unit-name.c:215
    #5 0x564ecaae6e7d in main ../src/test/test-unit-name.c:486
    #6 0x7fd6cb233009 in __libc_start_main (/lib64/libc.so.6+0x21009)

SUMMARY: AddressSanitizer: 2547 byte(s) leaked in 32 allocation(s).
-------
```

Reply:
Oops! Fixed.